src/java/org/apache/cassandra/locator/HttpSeedProvider.java

```diff
@@ -0,0 +1,135 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.locator;
+
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.cassandra.locator.HttpServiceConnector.resolveMetadataUrl;
+import static org.apache.cassandra.locator.HttpServiceConnector.resolveRequestTimeoutMs;
+
+public class HttpSeedProvider implements SeedProvider
+{
+    private static final Logger logger = LoggerFactory.getLogger(HttpSeedProvider.class);
+    private static final Pattern NEW_LINES_PATTERN = Pattern.compile("\\n");
+
+    static final String SEEDS_URL_PROPERTY = "seeds_url";
+    static final String REQUEST_HEADERS_PROPERTY = "request_headers";
+    static final String REQUEST_TIMEOUT_PROPERTY = "request_timeout";
+    static final String DEFAULT_REQUEST_TIMEOUT = "30s";
+
+    private final HttpServiceConnector serviceConnector;
+    private final Map<String, String> headers;
+
+    public HttpSeedProvider(Map<String, String> parameters)
+    {
+        this(getProperties(parameters));
+    }
+
+    public HttpSeedProvider(Properties properties)
+    {
+        serviceConnector = new HttpServiceConnector(resolveMetadataUrl(properties, SEEDS_URL_PROPERTY),
+                                                    resolveRequestTimeoutMs(properties,
+                                                                            REQUEST_TIMEOUT_PROPERTY,
+                                                                            DEFAULT_REQUEST_TIMEOUT));
+        headers = resolveHeaders(properties);
+    }
+
+    @Override
+    public List<InetAddressAndPort> getSeeds()
+    {
+        try
+        {
+            String response = serviceConnector.apiCall("", headers);
+
+            List<String> lines = Arrays.stream(NEW_LINES_PATTERN.split(response))
+                                       .map(String::trim)
+                                       .filter(s -> !s.isBlank())
+                                       .distinct()
+                                       .collect(toList());
+
+            List<InetAddressAndPort> seeds = new ArrayList<>(lines.size());
+
+            for (String line : lines)
+            {
+                try
+                {
+                    seeds.add(InetAddressAndPort.getByName(line));
```

Comment:
If the `line` is specified without a port e.g. `127.0.0.1` the default `7000` is used. Is this correct? I would not expect this from a user point of view for the HTTP seed provider.

Reply:
If port is part of a `line` here, it will be used, if `line` does not contain port, default port of 7000 will be used.